packages/cli/src/commands/generate/scaffold/scaffold.js

```diff
@@ -206,7 +206,17 @@ const componentFiles = async (name, scaffoldPath = '') => {
     DateTime: {
       displayFunction: 'timeTag',
     },
-    String: {
+    Int: {
+      name: 'NumberField',
+    },
+    Json: {
+      name: 'TextArea',
+      displayFunction: '',
+    },
+    Float: {
+      validation: "{{ required: true, type: 'Float' }}",
```

Comment:
How come this is the only one with an explicit `type` prop set and not also `Int`?

Reply:
When deciding what coercion to make, I look at two things; the name of the component used for the field, and the `type` prop. For `Int` I use a `<NumberField>` component, and those will be coerced using `parseInt`. For `Float` I use a regular `<TextField>`, for which no coercion is done at all by default. By adding `type: 'Float'` the `parseFloat` coercion will kick in.

Now, you might wonder why I use `<TextField>` for floats, and the reason for that is that `<NumberField>` is just a `<input type="number">` under the hood, and those don't support decimals by default. There is a way around that, by setting the `step` attribute to something < 1 (and > 0). For example `step="0.01"` will allow you to enter floats with two decimals, but not three or more decimals. I didn't want to enforce how many decimals our scaffolded forms support, so that's why I went with `<TextField>`. That way the user can enter however many decimals they want.

It might be a little misleading to have `type` be part of `validation` since I'm not actually using it for form validation. Should I move it to its own prop? And if so, maybe rename it to something like `dataType` to not confuse it with the `type` used on the underlying `<input>`

Another option might be to actually use it for validation by trying to build custom validation functions to pass to react-hook-form based on the type. If we want to go this route, could we do that as a separate PR?

Reply:
Makes sense! I like that idea of changing it to `dataType` so as not to confuse with regular `type` on the input. And it should probably move out of `validation` to keep it simple and we can pass that object in bulk to react-hook-form without having to tweak anything.

Yeah I think in the future we want to let users set `dataType` themselves on any field they want, including passing functions for custom coercion. We can definitely do that as another PR.